pkg/R/RDD.R

```diff
@@ -1294,6 +1294,79 @@ setMethod("sortBy",
             values(sortByKey(keyBy(rdd, func), ascending, numPartitions))
           })
 
+# Helper function to get first N elements from an RDD in the specified order.
+# Param:
+#   rdd An RDD.
+#   num Number of elements to return.
+#   ascending A flag to indicate whether the sorting is ascending or descending.
+# Return:
+#   A list of the first N elements from the RDD in the specified order.
+#
+takeOrderedElem <- function(rdd, num, ascending = TRUE) {          
+  if (num <= 0L) {
+    return(list())
+  }
+  
+  partitionFunc <- function(part) {
+    if (num < length(part)) {
+      # R limitation: order works only on primitive types!
+      ord <- order(unlist(part, recursive = FALSE), decreasing = !ascending)
+      part[ord[1:num]]
+    } else {
+      part
+    }
+  }
+  
+  newRdd <- mapPartitions(rdd, partitionFunc)
+  take(sortBy(newRdd, function(x) { x }, ascending = ascending), num)
```

Comment:
Unfortunately this doesn't quite fix it completely -- The `reduce` method in SparkR is implemented using a `collect` so this will get all `numPartitions * num` entries to the driver. FWIW I am fine with this approach for now.

It might be good to open a JIRA to fix `reduce` by calling `collect` on one partition at a time -- However that will reduce the amount of parallelism that we get and I'm not sure that is a good idea either (say a 1000 partition int RDD is being reduced using `+`, right now 1000 parts will be computed in parallel -- if we call collectPartition on one at a time then only 1 will be computed in parallel ?).